### PR TITLE
Update shader tests

### PR DESCRIPTION
**Pull request by Kangz**
@zhenyao PTAL, this fixes the preprocessor.html Chromium failures as well as part of the linkage tests. (the rest are addressed in the interface blocks CL).

### CODE REVIEW

**Comment by zhenyao** [comment]
Can you explain a bit why the change?  Every deviation from the native deqp should be documented (at least a comment in the test itself).

**Comment by Kangz** [comment]
Added notes to the preprocessor tests that were not updated. The other changes to preprocessor.test are  upstream changes.

**Comment by kenrussell** [comment]
LGTM assuming these match the upstream tests now but @zhenyao should re-review. Thanks for updating these tests.

**Comment by zhenyao** [comment]
Thank you. LGTM.

File: sdk/tests/deqp/data/gles3/shaders/preprocessor.test [hunk 1]
```diff
@@ -1647,7 +1647,6 @@ group conditional_inclusion "Conditional Inclusion Tests"
 
 	case basic_2
 		version 300 es
-		expect compile_fail
 		values { output float out0 = 1.0; }
```
Comment (qkmiao) [inline sdk/tests/deqp/data/gles3/shaders/preprocessor.test]: These changes may violate current WebGL 2.0 spec, see https://github.com/KhronosGroup/WebGL/pull/1523.
FYI: @Oletus

File: sdk/tests/deqp/data/gles3/shaders/linkage.test [hunk 2]
```diff
@@ -3494,43 +3607,6 @@ group uniform "Uniform linkage"
 	end
 
 	group block "Uniform blocks"
-		case precision_mismatch
```
Comment (qkmiao) [inline sdk/tests/deqp/data/gles3/shaders/linkage.test]: Chrome still fails on this test. So there should be precision mismatch bugs for uniform blocks. Should we move it to conformance2 if it doesn't exist in native dEQP?

**Pull request merged**